client.lua

```diff
@@ -194,32 +207,146 @@ local function StartFueling(vehicle)
     notify(fuel)
 end
 
-RegisterCommand('startfueling', function()
+local function StartFuelingWithCan(vehicle)
+    isFueling = true
     local ped = PlayerPedId()
 
-    if not inStation or isFueling or IsPedInAnyVehicle(ped) then
-        print('skipping fuel -- debug')
-        return
+    local Vehicle = Entity(vehicle).state
+    local fuel = Vehicle.fuel
+
+    if 100 - fuel < ox.refillValue then
+        isFueling = false
+        return notify('Tank full')
     end
 
-    local playerCoords = GetEntityCoords(ped)
+    local duration = math.ceil((100 - fuel) / ox.refillValue) * ox.refillTick
 
-    local isNearPump = findClosestPump(playerCoords)
+    TaskTurnPedToFaceEntity(ped, vehicle, duration)
+
+    Wait(500)
+
+    exports.ox_inventory:Progress({
+        duration = duration,
+        label = 'Fueling vehicle with Can',
+        useWhileDead = false,
+        canCancel = true,
+        disable = {
+            move = true,
+            car = true,
+            combat = true,
+        },
+        anim = {
+            dict = 'timetable@gardener@filling_can',
+            clip = 'gar_ig_5_filling_can',
+            flags = 49,
+        },
+    }, function(cancel)
+        if cancel then
+            isFueling = false
+        end
+    end)
+
+    while isFueling do
+
+        -- TODO: reduce can durability 
+
+        fuel += ox.refillValue
+
+        if(fuel >= 100) then
+            isFueling = false
+            fuel = 100.0
+        end
 
-    if not isNearPump then
-        return notify('Move closer to pump')
+        Wait(ox.refillTick)
     end
 
+    Vehicle:set('fuel', fuel, true)
+    SetVehicleFuelLevel(vehicle, fuel)
+    TriggerServerEvent('ox_fuel:pay', price)
+    -- DEBUG
+    notify(fuel)
+end
+
+local function GetPetrolCan(pumpCoord)
+    local ped = PlayerPedId()
+    local petrolCan = exports.ox_inventory:Search('count', 'WEAPON_PETROLCAN')
+
+    LocalPlayer.state.invBusy = true
+
+    TaskTurnPedToFaceCoord(ped, pumpCoord, ox.petrolCan.duration) 
+    -- Linden broke this changing from entity too coord, needs a better solution
+
+    Wait(500)
+
+    exports.ox_inventory:Progress({
+        duration = ox.petrolCan.duration,
+        label = 'Fueling can',
+        useWhileDead = false,
+        canCancel = true,
+        disable = {
+            move = true,
+            car = true,
+            combat = true,
+            mouse = false
```

Comment:
These don't disable movement in game for some reason...?

Reply:
I didn't touch these at all acutally, just copy pasted, now I notice `mouse = false` was removed so I will do the same

Reply:
well that's what's confusing me as this *should* work, I'd be weird if it does work if you add `mouse = false` lol

Reply:
These are actually working for me either way